### PR TITLE
ENH: special: add log trick to `roots_jacobi` and `roots_sh_jacobi`

### LINKED ISSUE #12425: roots_sh_jacobi: Error handling for OverflowError
**Is your feature request related to a problem? Please describe.**
I'm using `scipy.special.roots_sh_jacobi` to estimate an integral of a Beta RV for Bayesian A/B testing.
I have A/B test's with tens of thousands of users, but roots_sh_jacobi throws a OverflowError in those cases.
On the other hand, R's `statmod::gauss.quad.prob` is able to handle it without a problem.

R: `gauss.quad.prob(24, 'beta', alpha = 10, beta = 10)`
Py: `roots_sh_jacobi(24, 20 - 1, 10)`
Give the same answer

R: `gauss.quad.prob(24, 'beta', alpha = 10000, beta = 90000)`
py: `roots_sh_jacobi(24, 100000 - 1, 10000)`
OverflowError: (34, 'Result too large')

**Describe the solution you'd like**
in orthogonal.py l275: take the log of mu instead of mu itself
(e.g     `log_mu0 = (alpha+beta+1) *np.log(2.0) + cephes.betaln(alpha+1, beta+1)` )

in orthogonal.py l211: log trick again
(     log_w = np.log(w);  log_w += log_mu0 - log(w.sum());  w = np.exp(log_w) ) 

**Describe alternatives you've considered**
Im using the statmod package through the rpy2 bridge - which really sucks since it causes lots of dependencies

**Additional context (e.g. screenshots)**
I think the examples are enough to recreate :)

### PR DESCRIPTION
**Pull request by itamarfaran**
#### Reference issue
closes #12425: roots_sh_jacobi: Error handling for OverflowError

#### What does this implement/fix?
add option that allows returning of the log values in orthogonal._gen_roots_and_weights when symmetrize=False

#### Additional information
needed to imitate statmod::gauss.quad.prob function in R

### CODE REVIEW

File: scipy/special/orthogonal.py [hunk 1]
```diff
@@ -194,21 +194,27 @@ def _gen_roots_and_weights(n, mu0, an_func, bn_func, f, df, symmetrize, mu):
     dy /= np.abs(dy).max()
     w = 1.0 / (fm * dy)
 
+    if log_mu0 is not None:
+        if symmetrize: raise ValueError('cannot set symmetrize=True with log_mu0 not None')
+        
+        log_w = np.log(w) + log_mu0 - np.log(w.sum())
```
Comment (sethtroisi) [inline scipy/special/orthogonal.py:200]: when I tried this I saw negative w values. but I tried the log trick for all roots (not just roots_jacobi)

File: scipy/special/orthogonal.py [hunk 2]
```diff
@@ -194,21 +194,27 @@ def _gen_roots_and_weights(n, mu0, an_func, bn_func, f, df, symmetrize, mu):
...
-    else:
-        return x, w
+    if mu: return x, w, mu0
+    return x, w
 
 # Jacobi Polynomials 1               P^(alpha,beta)_n(x)
 
 
-def roots_jacobi(n, alpha, beta, mu=False):
+def roots_jacobi(n, alpha, beta, mu=False, log_values=False):
```
Comment (sethtroisi) [inline scipy/special/orthogonal.py:217]: This option should be tested.

**Review by sethtroisi — CHANGES_REQUESTED** [review]
I failing tests are 

FAILED scipy/special/tests/test_orthogonal.py::test_roots_jacobi - AssertionE...
FAILED scipy/special/tests/test_orthogonal.py::test_roots_sh_jacobi - Asserti...

So that needs to be looked into

File: scipy/special/orthogonal.py (hunk 1 above)
Comment (itamarfaran) [reply scipy/special/orthogonal.py]: should be more explicit now: when passing log_weights=True, the user *expects* to get the log of the weights, thus they are allowed to be negative

File: scipy/special/orthogonal.py (hunk 2 above)
Comment (itamarfaran) [reply scipy/special/orthogonal.py]: added test here (didn't push it yet), although I get the following failure in test_orthogonal.py#L331:

when log_weights=True, I'm getting an difference of .023, which is a relative difference of about 1.69e-15 (mu is of the scale of 10^13). So the test is a bit to harsh for taking exponents and logs. what should I do about that?

regarding the normal behavior (log_weights=False), all tests pass.

**Comment by itamarfaran** [comment]
the tests pass now for log_weights=False

File: scipy/special/orthogonal.py (hunk 2 above)
Comment (itamarfaran) [reply scipy/special/orthogonal.py]: see my comment in the discussion

**Comment by itamarfaran** [comment]
@person142 any new with this pr :)?

**Comment by person142** [comment]
Ach let me try to find some time to review... first thought though is that a flag determining whether we return weights or log weights is non-ideal API-wise; generally outputs of functions shouldn't change completely depending on flags. (Admittedly yes the API already does that, but we can no longer correct that because of back compat. Would be nice to not make the situation worse though.)

**Comment by itamarfaran** [comment]
@person142 would it be better to open a completely new module with 'roots_log_ ...' functions instead?

**Comment by person142** [comment]
> would it be better to open a completely new module with 'roots_log_ ...' functions instead?

Yes, new functions would be the way to go I think. It would match well with the rest of the API-e.g. `loggamma` versus `gamma`, `log_ndtr` versus `ndtr` and so on. But before making you put more effort into this, let me read https://github.com/scipy/scipy/issues/12425 more carefully first... might not have time for that until next week though.

**Pull request closed without merging**